osbs/utils.py

```diff
@@ -289,6 +289,10 @@ def get_time_from_rfc3339(rfc3339):
         return timegm(time_tuple)
 
 
+
+VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[a-z0-9]')
```

Comment:
If we still want to allow dashes and dots:
1) remove non allowed characters from repo and branch. Anything other than `[\-\.a-z0-9]' - We still want this because underscores, for instance, are not allowed.
2) strip and rstrip `'.'` and `'-'` from repo and branch

Let's agree on the algorithm before any further code changes.